toolkit/tools/internal/azureblobstorage/azureblobstorage.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package azureblobstoragepkg
+
+import (
+	"context"
+	"errors"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+)
+
+const (
+	AnonymousAccess = 0
+	AuthenticatedAccess = 1
+)
+
+type AzureBlobStorage struct {
+	theClient *azblob.Client
+}
+
+func (abs *AzureBlobStorage) Upload(
+	ctx context.Context,
+	localFileName string,
+	containerName string,
+	blobName string) (err error) {
+
+	uploadStartTime := time.Now()
+
+	localFile, err := os.OpenFile(localFileName, os.O_RDONLY, 0)
+	if err != nil {
+		logger.Log.Infof("  failed to open local file for upload. Error: %v", err)
+		return err
+	}
+	defer localFile.Close()
+
+	_, err = abs.theClient.UploadFile(ctx, containerName, blobName, localFile, nil)
+	if err != nil {
+		logger.Log.Infof("  failed to upload local file to blob. Error: %v", err)
+		return err
+	}
+
+	uploadEndTime := time.Now()
+	logger.Log.Infof("  upload time: %s", uploadEndTime.Sub(uploadStartTime))
+
+	return nil
+}
+
+func (abs *AzureBlobStorage) downloadInternal(
+	ctx context.Context,
+	containerName string,
+	blobName string,
+	localFileName string) (err error) {
+
+	downloadStartTime := time.Now()
+
+	localFile, err := os.Create(localFileName)
+	if err != nil {
+		logger.Log.Infof("  failed to create local file for download. Error: %v", err)
+		return err
+	}
+	defer localFile.Close()
```

Comment:
We ran into an issue with our network function were we would create an empty file, fail the download, and then leave the file. Might want a similar cleanup here. (see the `defer func()...` in `network.go:DownloadFile()`.

Reply:
I indeed ran into this problem while testing on a non-clean machine (dev scenario). I've solved it by having the Download() call downloadInternal() and then clean-up. I'll merge and use the `defer func()` pattern as it's more contained.

Reply:
Updated to match network.go:DownloadFile().